### PR TITLE
WIP, ENH: Make raw.plot_psd work for all channels

### PR DESCRIPTION
**Pull request by jeythekey**
This PR adds an optional argument to raw.plot_psd, which makes it possible to plot the psd for all kind of channels, including EMG (and e.g. behavioral data in misc), while keeping the actual behavior as default.
I don't know how useful this is for others, if it is deemed irrelevant, feel free to reject.

### CODE REVIEW

**Comment by agramfort** [comment]
I am -1 on this one. You can use the picks parameter. Also again this comes with no test. it's also a good practice to avoid having functions with too many parameters.

**Comment by jona-sassenhagen** [comment]
@jeythekey I think we all appreciate your code submissions, but it is usually better to first open an issue and discuss if we want to include it. MNE is becoming a bigger and bigger project, so we think a lot about backwards compatibility with existing code, and long-term code maintainability.

**Comment by jeythekey** [comment]
@agramfort, well, currently it is not possible to plot the psd of any channel types besides 'meg', 'eeg', 'seeg', 'ecog' with raw.plot_psd(_topo), which I find is i) inconsistent with the behavior of raw.plot() and is ii) kind of an arbitrary limitation, at least for plot_psd. The additional optional argument is only added to retain the current behavior as default, i.e. to ensure backward compatibility.
Sorry, that I did not open an issue first, I had done this change for myself before anyway and thought it makes thinks easier if discussions are not unnecessarily scattered.
If you are still -1, I will close this right away, of course.

**Comment by agramfort** [comment]
+1 to make evoked plot and raw plot psd more consistent in what channels they display. So maybe we have a consistency effort to make it. But I would not do this with a new parameter. Here you can achieve what you want with the picks params AFAIK. Shall we plot the same channel types in raw.plot_psd and evoked.plot? I am ok with this.

**Comment by jeythekey** [comment]
Yes, you are right, we could restrict to the brain ephys channels `if picks is None`, much better. Concerning consistency with the epochs-functions, I did not look into this yet. I will try to do this the next days.

**Comment by jeythekey** [comment]
So, I removed the parameter again. This is not ready yet, but I already wanted to see if any tests complain.
TODO:
- Check consistency with evoked.plot_... and epochs.plot_... concerning channels to plot
- Add test for plotting with picks=None (old behavior) and for picks='emg', e.g.

File: mne/viz/raw.py [hunk 1]
```diff
@@ -563,31 +564,48 @@ def _set_psd_plot_params(info, proj, picks, ax, area_mode):
...
         these_picks = np.intersect1d(these_picks, picks)
         if len(these_picks) > 0:
             picks_list.append(these_picks)
-            titles_list.append(titles[name])
-            units_list.append(units[name])
-            scalings_list.append(scalings[name])
+            try:
+                titles_list.append(titles[pick])
+            except KeyError:
+                titles_list.append(pick.upper())
```
Comment (larsoner) [inline mne/viz/raw.py]: These can be simplified using `dict.get`:
```
titles_list.append(titles.get(pick, pick.upper()))
```

**Comment by massich** [comment]
@jeythekey is this PR also addressing #5816 ?

**Comment by jeythekey** [comment]
@massich No, I think it is independent.

**Comment by larsoner** [comment]
@jeythekey interested in coming back to this?

**Comment by larsoner** [comment]
Closing due to non-response, but @jeythekey feel free to let me know if you're still interested in working on this
